libraries/AP_Scripting/docs/docs.lua

```diff
@@ -976,17 +976,59 @@ function winch:relax() end
 function winch:healthy() end
 
 -- desc
----@class mount
-mount = {}
+---@class camera
+camera = {}
 
 -- desc
----@param param1 integer
----@return integer|nil  -- pic_count
----@return boolean|nil  -- record_video
----@return integer|nil  -- zoom_step
----@return integer|nil  -- focus_step
----@return boolean|nil  -- auto_focus
-function mount:get_camera_state(param1) end
+---@param instance integer
+---@param distance_m number
+function camera:set_trigger_distance(instance, distance_m) end
+
+-- desc
+---@param instance integer
+---@param start_recording boolean
+---@return boolean
+function camera:record_video(instance, start_recording) end
+
+-- desc
+---@param instance integer
+function camera:take_picture(instance) end
+
+-- desc
+---@class camera_state_t_ud
+local camera_state_t_ud = {}
+
+---@return camera_state_t_ud
+function camera_state_t() end
```

Comment:
This is correct, but this is a example where we actually don't really need the constructor. Your not going to be declaring these in the script, just getting them as a return. We would need to change the binding generator to allow us to hide it.

Reply:
Discussed at the dev call and I think we agreed it was OK for us to leave this enhancement for a future PR.  I've raised an issue here though so that it's not so easily forgotten.  https://github.com/ArduPilot/ardupilot/issues/23445